File: life-agent-web/src/app/actions/lifeChat.ts
"use server";

import { getToken } from "./auth";

const API_BASE = process.env.API_BASE_URL || "http://localhost:5140";

export interface LifeChatResponse {
  success: boolean;
  response: string;
  usedEventCount: number;
  usedMemoryCount: number;
  readOnly: boolean;
  wroteData: boolean;
  executed: boolean;
}

export async function askLifeChat(message: string, clientTimeZone?: string): Promise<LifeChatResponse> {
  const token = await getToken();
  if (!token) throw new Error("Unauthorized");

  const res = await fetch(`${API_BASE}/api/life/chat`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({
      message,
      clientTimeZone: clientTimeZone || "Asia/Shanghai",
    }),
    cache: "no-store",
  });

  const data = await res.json();
  if (!res.ok) {
    throw new Error(data.error?.message || "暂时无法回答这个问题");
  }

  return data;
}

File: LifeAgent.Api/Models/LifeChat.cs
namespace LifeAgent.Api.Models;

public class LifeChatRequest
{
    public string Message { get; set; } = string.Empty;
    public string? ClientTimeZone { get; set; }
}

public class LifeChatResponse
{
    public bool Success { get; set; } = true;
    public string Response { get; set; } = string.Empty;
    public int UsedEventCount { get; set; }
    public int UsedMemoryCount { get; set; }
    public int UsedReminderCount { get; set; }
    public int UsedPlanSignalCount { get; set; }
    public LifeChatUsedContext UsedContext { get; set; } = new();
    public bool ReadOnly { get; set; } = true;
    public bool WroteData { get; set; } = false;
    public bool Executed { get; set; } = false;
}

public class LifeChatUsedContext
{
    public IReadOnlyList<LifeChatUsedContextItem> Items { get; set; } = Array.Empty<LifeChatUsedContextItem>();
    public bool ReadOnly { get; set; } = true;
    public bool WroteData { get; set; } = false;
    public bool Executed { get; set; } = false;
}

public class LifeChatUsedContextItem
{
    public string Id { get; set; } = string.Empty;
    public string SourceType { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Detail { get; set; } = string.Empty;
    public string Href { get; set; } = string.Empty;
    public string Reason { get; set; } = string.Empty;
}
